### PR TITLE
feat: make Plate-component draggable

### PR DESCRIPTION
**Pull request by simbig**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
   },
   "dependencies": {
     "@ant-design/icons": "5.0.1",
+    "@dnd-kit/core": "^6.0.8",
```
Comment (spawnia) [inline package.json:51]: How does this library compare to alternatives? What is its footprint in terms of bundle size?

File: src/Plate/wellUtils.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { PALETTE } from '../theme';
+
+export function generalWellStyle() {
```
Comment (spawnia) [inline src/Plate/wellUtils.ts:3]: can be a const, missing type

File: src/Plate/index.tsx [hunk 3]
```diff
@@ -23,57 +24,68 @@ export function Plate(props: PlateProps) {
...
-        <MllSpinnerIcon
-          style={{
-            width: '2em',
-          }}
-        />
-      }
+    <DndContext
+      collisionDetection={rectIntersection}
+      onDragEnd={(e) => {
+        if (props.onDragEnd) {
```
Comment (spawnia) [inline src/Plate/index.tsx:30]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls

File: src/Plate/index.stories.tsx [hunk 4]
```diff
@@ -57,6 +58,19 @@ const Template: Story<Partial<PlateProps>> = function Template(args) {
...
 };
 
+const DraggablePlateTemplate: Story<Partial<PlateProps>> =
+  function DraggablePlateTemplate(args) {
+    return (
+      <Plate
+        data={null}
+        {...args}
+        onDragEnd={
+          action('onDragEnd') // dataLocation: `const sourceData = e.active.data.current; const targetData = e.over?.data.current;`
```
Comment (spawnia) [inline src/Plate/index.stories.tsx:68]: ?

Comment (simbig) [reply src/Plate/index.stories.tsx:68]: A description of how a developer can get the data object of the source (draggable) and target (droppable).

File: package.json (hunk 1 above)
Comment (simbig) [reply package.json:51]: I did a bit of research, before choosing it.

https://npmtrends.com/@dnd-kit/core-vs-react-beautiful-dnd-vs-react-dnd-vs-react-drag-and-drop-vs-react-draggable

> the core of the library weighs around 10kb minified

> It is a zero-dependent, feature-rich dnd library

> Dnd-kit is a pretty good lightweight solution

Let me know if you have another recommendation.

File: src/Plate/wellUtils.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { PALETTE } from '../theme';
+
+export const GENERAL_WELL_STYLE: {
```
Comment (spawnia) [inline src/Plate/wellUtils.ts:3]: Better use `CSSProperties` as the type to avoid repetition

File: package.json (hunk 1 above)
Comment (spawnia) [reply package.json:51]: Looks good, thanks.

**Pull request merged**